compiler/x/codegen/DataSnippet.cpp

```diff
@@ -105,12 +105,21 @@ TR_Debug::print(TR::FILE *pOutFile, TR::IA32DataSnippet * snippet)
    printSnippetLabel(pOutFile, snippet->getSnippetLabel(), bufferPos, getName(snippet));
    printPrefix(pOutFile, NULL, bufferPos, snippet->getDataSize());
 
-   trfprintf(pOutFile, "%s \t%s", snippet->getDataSize() == 8 ?
-                 dqString() :
-                 (snippet->getDataSize() == 4 ?
-                  ddString() :
-                  dwString()),
-                 hexPrefixString());
+   switch (snippet->getDataSize())
+      {
+      case 8:
+         trfprintf(pOutFile, "%s \t%s", dqString(), hexPrefixString());
+         break;
+      case 4:
+         trfprintf(pOutFile, "%s \t%s", ddString(), hexPrefixString());
+         break;
+      case 2:
+         trfprintf(pOutFile, "%s \t%s", dwString(), hexPrefixString());
+         break;
+      default:
+         trfprintf(pOutFile, "%s \t%s", dbString(), hexPrefixString());
+         break;
+      }
```

Comment:
Could we save some DLL code size here by rewriting this something like:
```
const char *toString;
switch (snippet->getDataSize())
   {
   case 8:  toString = dqString(); break;
   case 4:  toString = ddString(); break;
   // etc
   }
trfprintf(pOutFile, "%s \t%s", toString, hexPrefixString());
```

Reply:
Sure, it has been updated.